crates/fuel-core/src/schema/block.rs

```diff
@@ -221,19 +223,26 @@ impl BlockQuery {
             (Some(id), None) => id.into(),
             (None, Some(height)) => {
                 let height: u64 = height.into();
-                db.get_block_id(height.try_into()?)?
-                    .ok_or("Block height non-existent")?
+                let data = BlockQueryContext(ctx.data_unchecked());
+                data.block_id(height).map_err(|_| {
+                    async_graphql::Error::new(format!(
+                        "Block with height {} not found",
+                        height
+                    ))
+                })?
             }
             (None, None) => {
                 return Err(async_graphql::Error::new("Missing either id or height"))
             }
         };
 
-        let block = db
-            .storage::<FuelBlocks>()
-            .get(&id)?
-            .map(|b| Block::from(b.into_owned()));
-        Ok(block)
+        let data = BlockQueryContext(ctx.data_unchecked());
+        let block = data.block(id)??;
+
+        Ok(Some(Block {
```

Comment:
I guess `block.into` should be enough. If not, we can add this kind of convertion=)

Reply:
`Into` is implemented for this tuple variant so we need to use a literal